clang/lib/Sema/SemaTemplateDeduction.cpp

```diff
@@ -3553,6 +3553,56 @@ static unsigned getPackIndexForParam(Sema &S,
   llvm_unreachable("parameter index would not be produced from template");
 }
 
+// if `Specialization` is a `CXXConstructorDecl` or `CXXConversionDecl`
+// we try to instantiate and update its explicit specifier after constraint
+// checking.
+static Sema::TemplateDeductionResult
+tryInstantiateExplicitSpecifier(Sema &S, FunctionDecl *Specialization,
```

Comment:
nit: Do you think we should make this function a (private) member of `Sema` instead of a static function?

Reply:
Since this function isn't reused, I prefer to leave it `static` :)